### PR TITLE
Milestone 2 - React & Redux Bookstore

### PR DESCRIPTION
**Pull request by sergioobolevich**
## In this milestone we have worked on the following tasks:

- Implement components: Book BookList and BooksForm.
- Define Redux actions creators and reducers (CREATE_BOOK and REMOVE_BOOK)
- Connect redux store with components.
- Made form to create books.
- Fix all errors.

Let us know of any corrections or suggestions.
Thanks!!

### CODE REVIEW

File: src/components/BooksList.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      <table className="table-auto">
+        <thead>
+          <tr>
+            <th>Book ID</th>
+            <th>Title</th>
+            <th>Category</th>
+          </tr>
+        </thead>
+        <tbody>
+          {books}
```
Comment (wathigo) [inline src/components/BooksList.js]: **REQUIRED**
* It is placed in `src/containers` directory. Let's separate presentational and container elements.

File: src/components/BooksForm.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { addBook } from '../actions';
+
+function BooksForm({ dispatch }) {
```
Comment (wathigo) [inline src/components/BooksForm.js]: Let place container and presentational components. Also, let's make use of es6 syntax.

File: src/components/BooksForm.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+  const categoryOptions = categories.map(category => (
+    <option key={category} value={category}>
+      {category}
+    </option>
+  ));
+
+  function handleChange(event) {
+    setTitle(event.target.value);
+  }
```
Comment (wathigo) [inline src/components/BooksForm.js]: Let's use object destructuring

File: src/components/BooksForm.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+  function handleSelect(event) {
+    setCategory(event.target.value);
+  }
```
Comment (wathigo) [inline src/components/BooksForm.js]: Let's use es6 syntax.
We can rewrite this as:
```js
const handleSelect = ({ target: {value}}) => {
  setCategory(value);
}
const

**Review by wathigo — CHANGES_REQUESTED** [review]
## CHANGES REQUIRED :o: 
Great job on this milestone!
Let's make a couple of changes before we proceed.

**After making the changes, submit another request for review.**

**Happy Coding :computer: 
Simon Wathigo**

File: src/components/BooksForm.js (hunk 2 above)
Comment (bellom) [reply src/components/BooksForm.js]: Hi Team
I think we still need to address this. `BooksForm` should also be in the container folder.
image

File: src/components/BooksForm.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
```
Comment (bellom) [inline src/components/BooksForm.js]: Hi Guys, Good job so far.
Let's do some rearrangement, just as specified in the previous review.

React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory, (presentational components) you have components that are not connected to the Redux store, and on your containers directory, you have components that are connected to the Redux store. [Read More...](https://redux.js.org/basics/usage-with-react#presentational-and-container-components) 

image
[Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/)

File: src/App.js [hunk 6]
```diff
@@ -1,9 +1,13 @@
 import React from 'react';
```
Comment (bellom) [inline src/App.js]: Let's move `app.js` inside the `component` folder.

image

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team  :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Review by teekaytech — APPROVED** [review]
# Status: Approved :green_circle: 

### Congratulations team :tada: :clinking_glasses: 

**This project has passed all the mandatory requirements :clap: 
It is therefore approved :heavy_check_mark:**

**Comment(s)**
- Going forward, you may consider improving the UI of your project, just to make it look more interesting and professional.
- You can also add form validations, so as to ensure no book is created when the fields are empty.

Again, congratulations, you can merge to the default branch.
Do not forget to fill the project completion form, as this is important for your weekly progress score.

**_Good luck with your next task._**

* * * 
**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Pull request merged**